### PR TITLE
Use fresher download stats from PyPI

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/chhantyal/py3readiness/issues/42.

This is similar to PR https://github.com/meshy/pythonwheels/pull/100.

I've made this, which is a weekly dump of the 5,000 most-downloaded packages from PyPI, using pypinfo:

* https://github.com/hugovk/top-pypi-packages
* https://hugovk.github.io/top-pypi-packages/

Now the new PyPI (Warehouse) has been released, `top_packages` has now been removed from the  API, here's a PR to instead use data from top-pypi-packages.

There's a choice of data over the preceding 30 days or 365 days. Python Wheels is going for 365 days.

Also in this PR:
* flake8 fixes
* Adds Python 3 compatibility
* Make runnable without S3 credentials
* Ensure flake8 and `make generate` work for Python 2 and 3 on Travis CI, for example: https://travis-ci.org/hugovk/py3readiness
* Update links to new PyPI

### CODE REVIEW

**Comment by chhantyal** [comment]
Hi @hugovk  Thank you for hard work. I really appreciate it.

However, I have to say I am not very comfortable about these changes. I understand we need to make changes as `top_packages` is going to be removed. But adding dependency on your git repo for fresh data also doesn't sound that reliable.

I will keep this PR open and will revisit when I have some time again.

**Comment by hugovk** [comment]
I'll split some of the non-top-pypi-packages specific commits into other PRs.

If you want to run pypinfo yourself to get the top packages list, have a look at how https://github.com/hugovk/top-pypi-packages does it, and let me know if I can help.

PS `top_packages` was removed from on Monday, but you may still be able to access it via http://legacy.pypi.org/ until 30 April 2018: https://wiki.python.org/psf/WarehouseRoadmap

**Comment by chhantyal** [comment]
Would be nice to run that `pypinfo` as part of generate run. Looks like it gets info from BigQuery, which means we have to sign up there. Not sure how much that costs.

**Comment by hugovk** [comment]
The first TB of queried BigQuery data each month is free. Each additional TB is $5.

If you don't sign up for billing, when you hit the TB limit you just get an error and no charge.

pypinfo tells you the estimated cost of a query; it costs an estimated $0.04 and 6.7 GB for the one for https://github.com/hugovk/top-pypi-packages/blob/master/top-pypi-packages-30-days.json

The quota is a rolling quota. That means it's not 1 TB per calendar month. If you hit the limit in the middle of the month, just wait a day or two and you'll have some quota freed up.

**Comment by chhantyal** [comment]
Ok, that's fair then. If you can adjust this PR to use `pypinfo` here instead of `wget` to your repo, that would be enough I guess.

So we will run approx. 4 times per month, without signup & it should be fine.

**Comment by hugovk** [comment]
You still need to sign up an account with Google BigQuery to be able to use pypinfo, with or without billing. See https://github.com/ofek/pypinfo#installation.

I set up https://github.com/hugovk/top-pypi-packages precisely to avoid the pain of setting it up local fetching :)

But if you prefer to do it manually, feel free to copy yourself what that repo is doing (see for example [generate.sh](https://github.com/hugovk/top-pypi-packages/blob/master/generate.sh) and the [server setup notes](https://github.com/hugovk/top-pypi-packages#server-setup-notes)). I can't really do it, as you would need to set it up with your pypinfo credentials.

**Comment by chhantyal** [comment]
ok, I will check it. We can call `pypinfo` programmatically from `generate.py`. Thus no need to setup cron job. I will check when I am free.

Thanks a ton for idea 👍

**Comment by chhantyal** [comment]
Hi @hugovk,
I have an simpler idea. Let's commit the 360 json data within this repo as static file (`data.json`) as it doesn't change that often.  

Then we just say something like:
``
Last updated on 10.08.2018
``

We can manually update `data.json` when there are some changes in 360 list. I think this will require very low maintenance. What do you think?

**Comment by hugovk** [comment]
Sure, I've updated this PR.

So the update process will be something like:

```sh
# download new data.json, regenerate results.json and wheel.svg
make generate

# update repo
git add data.json results.json wheel.svg
git commit -m "Update data"
git push
```
The last updated date is automatically read from the JSON file, just need to remove the "Updated weekly." bit.

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,7 @@ help:
 	@echo "make update   -- upload the json and index.html to s3"
 
 generate:
+	wget https://hugovk.github.io/top-pypi-packages/top-pypi-packages-365-days.min.json -O data.json
```
Comment (chhantyal) [inline Makefile]: Looks very good. Just few questions here:
- Looks like we will have top packages since last year?
- How many packages does it include?

Comment (hugovk) [reply Makefile]: Yes, it's get from the last 365 days. There's also an option for the last 30 days. Would you prefer that?

There's 5,000 packages listed in the file. It's about 240K in size. Actually, that file doesn't need to be committed to this repo, it's only needed as input to make results.json. Shall I remove `data.json` and ignore it?

Comment (chhantyal) [reply Makefile]: yea, 365 days is fine. In past, it was using even stats available for even longer time. But 365 days is probably better.

You are right, we don't need `data.json` file in the repo. Let's remove it & put on gitignore.

Comment (hugovk) [reply Makefile]: Done!

**Comment by chhantyal** [comment]
Thank you @hugovk for your contribution. Greatly appreciate it.
Site is now updated with your changes.

**Pull request merged**